requests-for-comments/2022-09-08-notifications.md

```diff
@@ -0,0 +1,81 @@
+# Request for comments: Notifications & @pauldambra 
+
+## Problem statement
+*Who are we building for, what are their needs (include example cases), why is this important?*
+
+Users don't know when to come back to PostHog. Something important can happen and they might miss it. This can be important things in the data, or that another user has edited your insight.
+
+## Success criteria
+*How do we know if this is successful (i.e. metrics, customer feedback), what's out of scope, whats makes this ambitious?*
+
+We have users from [the August 2022 customer interviews](https://github.com/PostHog/posthog/issues/11368) that we can revisit.
+
+We can measure:
+
+* number of triggers created
+* number of notifications sent
+* stickiness of dashboards and insights
+
+## Context
+*What are our competitors doing, what are the technical constraints, what are customers asking for, what does the data tell us, are there external motivations (e.g. launch week, enterprise contract)?*
+
+## Design 
+*What are the key user experience and technical design decisions / trade-offs?*
+
+```mermaid
+graph LR
+    A[Ingestion/Celery] -->|property seen| CS
+    A -->|read triggers| N
+    W-->|"create subscription (time-based trigger)"|N[("Trigger store (subscriptions)")]
+    W[Web] -->|create a trigger| N
+    Ac[Activity Log] -->|notable activity|CS
+    N ---|evaluate triggers| CS[Celery sender]
+    CS-->Slack
+    CS-->Email
+    Web ---|show notification list| NS
+    CS -->NS[(notification store)]
+```
+
+There are some designs in figma for comment https://www.figma.com/file/Y9G24U4r04nEjIDGIEGuKI/PostHog-Design-System-One?node-id=3112%3A1296
+
+### user considerations
+
+* how to create the triggers for notifications
+* where to see configured triggers
+* linking from notification to appropriate onward journey
+
+* expectation of performance
+  * e.g. evaluate whether insight series have crossed a threshold could be very expensive
+
+### Notifications people mentioned
+
+* chart goes above or below a limit
+    * expected to set this on an insight
+* for a single event if the value of a property is over a threshold
+* on particular calculated event
+* notification when same event occurs for a single user/group over time (cohort entry/exit?)
+* notification when experiment reaches significance
+
+### Tech considerations
```

Comment:
I'm guessing the core here is re-using subscriptions infra to send notifications as well to slack, not just images?

And the new bit is the notification store for in-app notifications, which are also triggered via celery?

Q: Can we re-use this to send custom in-app notifications? In the future, we might just want to use this to point users to something new ourselves ( see a very WIP dump: https://github.com/PostHog/product-internal/pull/346/files - i.e. it's a notification from posthog vs for a trigger you setup )

Just mentioning this so we know to keep in-app notifications not tightly coupled with the rest. It's unclear to me from the diagram that this is the case.